packages/flutter_tools/lib/src/build_system/targets/assets.dart

```diff
@@ -34,7 +34,7 @@ Future<Depfile> copyAssets(Environment environment, Directory outputDirectory) a
     assetBundle.entries.entries.map<Future<void>>((MapEntry<String, DevFSContent> entry) async {
       final PoolResource resource = await pool.request();
       try {
-        final File file = fs.file(outputDirectory.uri.resolve(entry.key));
+        final File file = fs.file(fs.path.join(outputDirectory.path, entry.key));
```

Comment:
The test is great, but this line and the one in bundle.dart should have some explanatory comments.

Reply:
I'll open another PR.